src/CognitiveServices/CognitiveServices/CognitiveServicesAccount/NewAzureCognitiveServicesCommitmentPlan.cs

```diff
@@ -0,0 +1,114 @@
+using Microsoft.Azure.Commands.Management.CognitiveServices.Properties;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.CognitiveServices;
+using Microsoft.Azure.Management.CognitiveServices.Models;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Management.Automation;
+using CognitiveServicesModels = Microsoft.Azure.Commands.Management.CognitiveServices.Models;
+
+namespace Microsoft.Azure.Commands.Management.CognitiveServices
+{
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "CognitiveServicesCommitmentPlan", SupportsShouldProcess = true), OutputType(typeof(CognitiveServicesModels.PSCognitiveServicesAccount))]
+    public class NewAzureCognitiveServicesCommitmentPlanCommand : CognitiveServicesAccountBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Resource Group Name.")]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Cognitive Services Account Name.")]
+        [ValidateNotNullOrEmpty]
+        [ValidatePattern("^[a-zA-Z0-9][a-zA-Z0-9_.-]*$")]
+        [ValidateLength(2, 64)]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Cognitive Services Account/CommitmentPlan Type.")]
+        [Alias(KindAlias)]
+        [AccountTypeCompleter()]
+        public string Type { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Cognitive Services Account/CommitmentPlan Sku Name.")]
+        [AccountSkuCompleter()]
+        public string SkuName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Cognitive Services CommitmentPlan Location.")]
+        [LocationCompleter("Microsoft.CognitiveServices/accounts")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Cognitive Services CommitmentPlan Tags.")]
+        [Alias(TagsAlias)]
+        [ValidateNotNull]
+        [AllowEmptyCollection]
+        public Hashtable[] Tag { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Cognitive Services CommitmentPlan Properties.")]
+        public CommitmentPlanProperties Properties { get; set; }
```

Comment:
```suggestion
        public CommitmentPlanProperties Property { get; set; }
```
- Keep singularity
- Suggest to wrap CommitmentPlanProperties with psobject to provide a way to avoid future breaking change.

Reply:
Can we keep using "Properties". I understand that singularity is preferred in here but we already have many commands using "Properties" instead of "Property" i think better keep consistent inside CS commandlets.

Reply:
Could you provide example used Properties?

Reply:
New-AzCognitiveServicesAccountCommitmentPlan
New-AzCognitiveServicesAccountDeployment

Reply:
fine